src/main/java/com/epam/aidial/core/controller/DeploymentController.java

```diff
@@ -100,13 +100,24 @@ private static DeploymentData createDeployment(Model model) {
     }
 
     static FeaturesData createFeatures(Features features) {
+        FeaturesData data = new FeaturesData();
+
         if (features == null) {
-            return null;
+            return data;
         }
-        FeaturesData data = new FeaturesData();
+
         data.setRate(features.getRateEndpoint() != null);
         data.setTokenize(features.getTokenizeEndpoint() != null);
         data.setTruncatePrompt(features.getTruncatePromptEndpoint() != null);
+
+        data.setSystemPrompt(booleanOrDefault(features.getSystemPromptSupported(), true));
+        data.setTools(booleanOrDefault(features.getToolsSupported(), false));
+        data.setSeed(booleanOrDefault(features.getSeedSupported(), false));
+
         return data;
     }
+
+    public static boolean booleanOrDefault(Boolean obj, boolean def) {
```

Comment:
Nit. Instead of introducing a new method we can update those fields conditionally.

Reply:
Fixed